utils/extract.go

```diff
@@ -10,65 +10,90 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
 )
 
+func sanitize(path string) (string, error) {
+	// Start by trimming redundant trailing slashes.
+	normalizedPath := strings.TrimRight(path, "/")
+
+	// Then anchor at '/' to allow us to leverage filepath.Clean's behaviour of stripping all
+	// instances of '../' for rooted paths. Note that we intentionally avoid filepath.Join
+	// here since it would call Clean prematurely.
+	if !strings.HasPrefix(normalizedPath, "/") {
```

Comment:
Paths that are already rooted should really be rejected, too. GNU tar doesn't allow extracting these IIRC, so it would be standard behavior.

Reply:
Since we always anchor this relative to the path at which we're unpacking, do we need to reject these outright? My understanding of tar's behaviour here was that it always unpacked relative to the output directory even for absolute paths, and didn't reject them outright

Reply:
Could be yeah. Should be fine even if we don't reject it here.